src/components/visualisation-bar/index.js

```diff
@@ -0,0 +1,246 @@
+import { select, div, span, h } from 'utils/selection';
+import { mergeDefined } from 'utils/utils';
+
+const wrapper = 'wrapper',
+    fill = 'fill',
+    label = 'label',
+    progress = 'progress',
+    labelText = 'label-text',
+    labelPc = 'label-percents';
+
+const elementsGroup = title => new Map([
+    [wrapper, div(`hx-visualisation-bar-${title} hx-visualisation-bar-p-wrapper`)],
+    [fill, div(`hx-visualisation-bar-fill hx-visualisation-bar-${title}-fill`)],
+    [progress, span(`hx-visualisation-bar-${title}-info hx-visualisation-bar-${title}-progress-info`)],
+    [label, div(`hx-visualisation-bar-label hx-visualisation-bar-${title}-label`)],
+    [labelPc, span('hx-visualisation-bar-label-percent')],
+    [labelText, span('hx-visualisation-bar-label-text')],
+
+])
+
+/**
+ * 
+ */
+export class VisualisationBar {
+    constructor(selector, options) {
+        this.__progress = 0;
+        this.__buffer = 0;
+
+        this.options = mergeDefined({
+            value: 0,
+            max: undefined,
+            buffer: 0,
+            valid: true,
+            content: '',
+            mock: false,
+            disabled: false,
+            withPercent: false
+        }, options);
+
+        this.selection = select(selector).classed('hx-visualisation-bar', true);
+        this.header = div('hx-visualisation-bar-header');
+        this.titleElement = h(3);
+        this.countElement = h(3);
+
+        this.body = div('hx-visualisation-bar-body hx-animate');
+        this.fillWrapper = div('hx-visualisation-bar-fill-wrapper');
+        this.labelWrapper = div('hx-visualisation-bar-label-wrapper');
+        this.progressElementsMap = elementsGroup('progress');
+        this.bufferElementsMap = elementsGroup('buffer');
+        this.balanceElementsMap = elementsGroup('balance');
+
+        this.contentElement = div('hx-visualisation-bar-content');
+
+        this.selection.add(this.header);
+        this.header.add(div('hx-visualisation-bar-title').add(this.titleElement));
+        this.header.add(div('hx-visualisation-bar-count').add(this.countElement));
+        this.selection.add(this.body);
+        this.body.add(this.fillWrapper).add(this.labelWrapper);
+
+        this.initFill();
+        this.initLabels();
+
+        this.selection.add(this.contentElement);
+        this.selection.api('visualisation-bar', this).api(this);
+        this.title = this.options.title;
+        this.progress = this.options.value;
+        this.buffer = this.options.buffer;
+        this.content = this.options.content;
+        this.disabled = this.options.disabled;
+        this.valid = this.options.valid;
+
+        if (this.options.mock && this.options.valid && !this.options.disabled) {
+            setInterval(() => {
+                this.progress = Math.round(Math.random() * 20);
+            }, 1000);
+        }
```

Comment:
This option and code should not be in the component. Updating based on an interval should live in the Demo or docs instead